diff --git a/client/src/app/constants/default-parameters.ts b/client/src/app/constants/default-parameters.ts
index ac8e462..73b8cf7 100644
--- a/client/src/app/constants/default-parameters.ts
+++ b/client/src/app/constants/default-parameters.ts
@@ -10,7 +10,7 @@ export const enum DefaultQueryParams {
   DEFAULT_NEIGHBORS    = 10,
   DEFAULT_MATCHED      = 4,
   DEFAULT_INTERMEDIATE = 5,
-  DEFAULT_SOURCE       = 'lis'
+  DEFAULT_SOURCE       = 'bipaa'
 }
 
 // alignment parameters
diff --git a/client/src/app/services/macro-tracks.service.ts b/client/src/app/services/macro-tracks.service.ts
index 987ac64..df12328 100644
--- a/client/src/app/services/macro-tracks.service.ts
+++ b/client/src/app/services/macro-tracks.service.ts
@@ -43,6 +43,7 @@ export class MacroTracksService {
   getChromosome(
     source: string,
     chromosome: string,
+    species_id: number,
     success = e => {},
     failure = e => {}
   ): void {
@@ -51,7 +52,7 @@ export class MacroTracksService {
     if (idx != -1) {
       let s: Server = AppConfig.SERVERS[idx];
       if (s.hasOwnProperty('chromosome')) {
-        let args = {chromosome: chromosome} as RequestOptionsArgs;
+        let args = {chromosome: chromosome, species_id: species_id} as RequestOptionsArgs;
         let response: Observable<Response>;
         if (s.chromosome.type === GET)
           response = this._http.get(s.chromosome.url, args)
