diff -Naur ../lis_context_viewer/client/src/app/components/multi/multi.component.ts /tmp/lis_context_viewer_fixed/client/src/app/components/multi/multi.component.ts
--- a/client/src/app/components/multi/multi.component.ts	2020-04-07 16:44:16.253851560 +0200
+++ b/client/src/app/components/multi/multi.component.ts	2020-04-07 16:43:55.343801614 +0200
@@ -8,7 +8,7 @@
          ViewChild,
          ViewEncapsulation }      from '@angular/core';
 import { Observable }             from 'rxjs/Observable';
-import Split                 from 'split.js';
+import * as Split                 from 'split.js';
 import * as d3                    from 'd3';
 import { GCV }                    from '../../../assets/js/gcv';
 
diff -Naur ../lis_context_viewer/client/src/app/components/search/search.component.ts /tmp/lis_context_viewer_fixed/client/src/app/components/search/search.component.ts
--- a/client/src/app/components/search/search.component.ts	2020-04-07 16:44:16.253851560 +0200
+++ b/client/src/app/components/search/search.component.ts	2020-04-07 16:43:55.343801614 +0200
@@ -10,7 +10,7 @@
          ViewChildren,
          ViewEncapsulation }      from '@angular/core';
 import { Observable }             from 'rxjs/Observable';
-import Split                 from 'split.js';
+import * as Split                 from 'split.js';
 import * as d3                    from 'd3';
 import { GCV }                    from '../../../assets/js/gcv';
 
